### DISCUSSION TITLE
Activate the trigger when the values in the useForm are updated

### DISCUSSION THREAD

**Discussion by MatthieuPombet** [react-hook-form/react-hook-form#9822] [Q&A]
Hi,

I work on a complex form with redux. The new props values is perfect for me, but when I update my store, I want to trigger my form. When the store updated, I activate the trigger with a useEffect on the store and I have a feeling that the trigger is activated before the update of the useForm and the response is always true. Do you have an idea for waiting the update of the useForm (values) without setTimeout ?

**Comment by bluebill1049** [comment]
can you do that at `onChange`?

**Reply by MatthieuPombet** [reply]
It is complicated, I use useFormContext and the trigger is in the parent file. If I use the trigger in the children, the validation will be on the useForm or on the useFormContext ? It could be a solution if the trigger valid the useForm. But I have a second problem, a part of my data come to an API called by another field. This complexify the use of onChange, I think
